packages/zoe/test/unitTests/contractSupport/test-offerTo.js

```diff
@@ -0,0 +1,246 @@
+// @ts-check
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+
+import { E } from '@agoric/eventual-send';
+import bundleSource from '@agoric/bundle-source';
+
+import { setup } from '../setupBasicMints';
+import { makeZoe } from '../../..';
+import { makeFakeVatAdmin } from '../../../src/contractFacet/fakeVatAdmin';
+import {
+  offerTo,
+  assertProposalShape,
+  swapExact,
+} from '../../../src/contractSupport/zoeHelpers';
+
+const contractRoot = `${__dirname}/../zcf/zcfTesterContract`;
+
+const makeOffer = async (zoe, zcf, proposal, payments) => {
+  let zcfSeat;
+  const getSeat = seat => {
+    zcfSeat = seat;
+  };
+  const invitation = await zcf.makeInvitation(getSeat, 'seat');
+  const userSeat = await E(zoe).offer(invitation, proposal, payments);
+  return { zcfSeat, userSeat };
+};
+
+const setupContract = async (moolaIssuer, bucksIssuer) => {
+  const instanceToZCF = new Map();
+  const setJig = jig => {
+    instanceToZCF.set(jig.instance, jig.zcf);
+  };
+  const zoe = makeZoe(makeFakeVatAdmin(setJig).admin);
+
+  // pack the contract
+  const bundle = await bundleSource(contractRoot);
+  // install the contract
+  const installation = await zoe.install(bundle);
+
+  // Create TWO instances of the zcfTesterContract which have
+  // different keywords
+  const issuerKeywordRecord1 = harden({
+    TokenA: moolaIssuer,
+    TokenB: bucksIssuer,
+  });
+
+  const issuerKeywordRecord2 = harden({
+    TokenC: moolaIssuer,
+    TokenD: bucksIssuer,
+  });
+
+  // contract instance A
```

Comment:
It would be better to not re-use A and B for both keywords and contracts, particularly since contractB uses keywords C and D.

Reply:
Good point. Let me see what I can do.

Reply:
Done